From 34f5fb0edf50013307a722d24556cdae45aa3346 Mon Sep 17 00:00:00 2001
From: Lu Yahan <yahan@iscas.ac.cn>
Date: Mon, 30 Oct 2023 17:57:30 +0000
Subject: [PATCH] [riscv] Add riscv B extension compilation flag

Change-Id: I1a7077338016c12cc4279a6c1d28074732803571
Reviewed-on: https://chromium-review.googlesource.com/c/chromium/src/+/4974179
Reviewed-by: Dirk Pranke <dpranke@google.com>
Commit-Queue: Dirk Pranke <dpranke@google.com>
Auto-Submit: Yahan Lu <yahan@iscas.ac.cn>
Cr-Commit-Position: refs/heads/main@{#1217051}
---

diff --git a/build/config/riscv.gni b/build/config/riscv.gni
index 5a78490..595b461 100644
--- a/build/config/riscv.gni
+++ b/build/config/riscv.gni
@@ -16,5 +16,16 @@
     #   512
     #   1024
     riscv_rvv_vlen = 128
+
+    # RISCV profile compilation flag. Possible values are:
+    # rv64gc
+    # rvau22
+    riscv_profile = "rv64gc"
+
+    # RISCV B extension compilation flag.
+    # See https://github.com/riscv/riscv-bitmanip/blob/main/bitmanip/bitmanip.adoc#colophon
+    riscv_use_zba = false
+    riscv_use_zbb = false
+    riscv_use_zbs = false
   }
 }
